### ISSUE TITLE
Bug: USDC Geometry Load Fails

### ISSUE THREAD

**Issue by AMZN-Gene** [assimp/assimp#5703]
Assimp::Importer::ReadFile() for a simple USD wooden spoon fails to load.

**To Reproduce**
1. Download [kitchen_set example from the OpenUSD website](https://openusd.org/release/dl_kitchen_set.html) (Note: we won't need the entire kitchen set)
2. Try to load any of the child usd geometry files (.geom.usd)
Example: `aiScene* myScene= Assimp::Importer::ReadFile(<path_to_Kitchen_set>\assetsWoodenSpoon.geom.usd)`
3. Notice _myScene_  = nullptr
    `Validation failed: aiScene::mNumMeshes is 0.  At least one mesh must be there.`

**Expected behavior**
The wooden spoon mesh is successfully loaded

**Platform (please complete the following information):**
 - OS: Windows
 - Version  11

**Additional context**
I was originally thinking that maybe USD was the wrong file extension since it's actually a usdc. 
Changing the woodenspoon.geom.usd to woodenspoon.geom.usdc also doesn't work.

**Comment by tellypresence** [comment]
Kitchen set not supported as mentioned in [issue 5547](https://github.com/assimp/assimp/issues/5547#issuecomment-2065552410), likely due to using "Composition" (layered/hierarchical)

**Comment by AMZN-Gene** [comment]
Thanks for quick reply! Even though hierarchies aren't implemented would that effect child nodes (such as the _woodenspoon.geom.usd_) to fail loading? I was thinking that the _woodenspoon.geom.usd_ doesn't rely on other usd files and is just a usdc storing vertex/mesh data.

**Comment by tellypresence** [comment]
One would hope so...but during my testing I couldn't get any of the sub-models to load from the kitchen set, which is consistent with your observation

**Comment by AMZN-Gene** [comment]
At the very least the error seems wrong. "No suitable reader found for the file format".
USD is a supported file format, I'm guessing the reader is just having particular trouble parsing it.

**Comment by AMZN-Gene** [comment]
Also interesting regarding hierarchies, I'm using the [O3DE game engine](https://github.com/o3de/o3de) AssImp integration. 
If I load kitchen set using AssImp, O3DE is able to generate the exact kitchen entities hierarchy. IE, I get a kitchen entity, with all its sub-entities (chairs, spoons, etc). So it's as in the hierarchy works great, but just the individual vertices aren't loaded.

**Comment by tellypresence** [comment]
In my case the USD loader is correctly invoked for all 3 models:
- `WoodenSpoon.usd`
- `WoodenSpoon.geom.usd`
- `WoodenSpoon_payload.usd`

But I get the same error message for all 3 models:
```
Validation failed: aiScene::mNumMeshes is 0.  At least one mesh must be there
```

**Comment by tellypresence** [comment]
(But please consider closing this issue since it's a known problem with an external project)

**Comment by AMZN-Gene** [comment]
I think this bug still stands (regardless of AssImp error), considering WoodenSpoon.geom.usd does contain a mesh. I can open it in Blender. I'll update the issue description's to include `Validation failed: aiScene::mNumMeshes is 0.  At least one mesh must be there`

**Comment by AMZN-Gene** [comment]
Any suggestions on which files/classes one would start looking at figure out how to parse the mesh?

**Comment by tellypresence** [comment]
Would start with the merged PR #5628

(Though I would suggest first going through the tinyusdz project and read up on any mentions of "kitchen set" to avoid disappointment...)

**Comment by AMZN-Gene** [comment]
Nice comment from @syoyo suggesting how to enable tiny's composition features for testing with kitchen set.

Strangely, even directly loading the geom assets (like assets/woodenspoon/woodenspoon.geom.usd) tydra::ListPrims(stage, meshmap) is failing to find any tinyusdz::GeomMesh's.
However, I can clearly see output printing the faces and vertices.
![image](https://github.com/user-attachments/assets/cf4f39bb-6d9e-4133-8010-63e6d5f0afc6)

**Comment by tellypresence** [comment]
The saga continues on [tinyusdz issue 173](https://github.com/lighttransport/tinyusdz/issues/173)

**Comment by AMZN-Gene** [comment]
Closing as blocked by TinyUSDZ:
https://github.com/lighttransport/tinyusdz/issues/173
https://github.com/lighttransport/tinyusdz/issues/94
https://github.com/lighttransport/tinyusdz/issues/169

Note: I've played around more with TinyUSDZ and it's composition options. Still no go.

I also updated TinyUSDZ's tydra::ListPrims() to do a more exhaustive search and it was able to detect the meshes, but attempting the access the vertices causes a crash, so there's more work to be done.
